### PR TITLE
refactor: Create IPathResolver service to simplify code that needs path information (#117)

### PR DESCRIPTION
**Pull request by rjstone**
Code refactoring for issue #117 

The code for resolving paths has been rather messy, often requiring like 2-3 different dependencies just to get the install path for a game. It's been kind of a mess. `IPathResolver` tries to encapsulate all of the complexity of resolving paths in one place so the rest of the code is simplified.

Testing: (This can wait until at least one feature is added.)
- Pretty much anything that might use a path will need to be tested for example installing the debug plugin, using the language server, project file generation, pyro builds, etc.
- Basically this touches lots of stuff. I will update with specific tests later.

Checklist before doing the merge commit:
- [] Review code in PR
- [] Testing
- [] Check format of title for CI (make sure it has valid `<type>: <subject>` format)
- [] **Use Squash and Commit**

### CODE REVIEW

**Comment by rjstone** [comment]
Scrivener found a bug with the welcome file path but this is fixed in #122 just to make merging easier. A header prefix with refactor shouldn't trigger distribution so this should be safe to merge.

**Comment by Scrivener07** [comment]
Ill be merging this to master with no CI release for the logistics of making new branches.
